decode-string.py

```diff
@@ -1,25 +1,25 @@
-class Solution:
-    def decodeString(self, s: str) -> str:
-        stack = []
-    
-        for i in s:
-            if i == ']':
-                letters = []
-                while stack and stack[-1] != '[':
-                    letters.append(stack.pop())
-                stack.pop()
-                num = []
-                while stack and stack[-1].isdigit():
-                    num.append(stack.pop())
-                new = letters[::-1] * int(''.join(num[::-1]))
```

Comment:
a lot happened in this one line, It's better if you add comment what is  heppening